l5kit/l5kit/visualization/visualiser/visualiser.py

```diff
@@ -0,0 +1,133 @@
+                trajectory_super_dict[trajectory.legend_label][k].append(v)
+
+        frame_dict = dict(ego=ColumnDataSource(ego_dict), agents=ColumnDataSource(agents_dict),
+                          lanes=ColumnDataSource(lanes_dict),
+                          crosswalks=ColumnDataSource(crosswalk_dict))
+        frame_dict.update({k: ColumnDataSource(v) for k, v in trajectory_super_dict.items()})
+
+        out.append(frame_dict)
+
+    f = bokeh.plotting.figure(
+        title="Scene {}".format(scene_index),
+        match_aspect=True,
+        x_range=(out[0]["ego"].data["center_x"][0] - 50, out[0]["ego"].data["center_x"][0] + 50),
+        y_range=(out[0]["ego"].data["center_y"][0] - 50, out[0]["ego"].data["center_y"][0] + 50),
+        tools=["pan", "wheel_zoom", agent_hover, "save", "reset"],
+        active_scroll="wheel_zoom",
+    )
+
+    f.xgrid.grid_line_color = None
+    f.ygrid.grid_line_color = None
+
+    f.patches(line_width=0, alpha=0.5, color="color", source=out[0]["lanes"])
+    f.patches(line_width=0, alpha=0.5, color="#B5B50D", source=out[0]["crosswalks"])
+    f.patches(line_width=2, color="#B53331", source=out[0]["ego"])
+    f.patches(line_width=2, color="color", name="agents", source=out[0]["agents"])
+
+    js_string = """
+            sources["lanes"].data = frames[cb_obj.value]["lanes"].data;
+            sources["crosswalks"].data = frames[cb_obj.value]["crosswalks"].data;
+            sources["agents"].data = frames[cb_obj.value]["agents"].data;
+            sources["ego"].data = frames[cb_obj.value]["ego"].data;
+
+            figure.x_range.setv({"start": frames[cb_obj.value]["ego"].data["center_x"][0]-50, "end": frames[cb_obj.value]["ego"].data["center_x"][0]+50})
+            figure.y_range.setv({"start": frames[cb_obj.value]["ego"].data["center_y"][0]-50, "end": frames[cb_obj.value]["ego"].data["center_y"][0]+50})
+
+            sources["lanes"].change.emit();
+            sources["crosswalks"].change.emit();
+            sources["agents"].change.emit();
+            sources["ego"].change.emit();
+        """
+
+    for trajectory_name in trajectories_labels:
+        f.multi_line(alpha=0.8, line_width=3, source=out[0][trajectory_name], color="color",
+                     legend_label=trajectory_name)
+        js_string += f'sources["{trajectory_name}"].data = frames[cb_obj.value]["{trajectory_name}"].data;\n' \
+                     f'sources["{trajectory_name}"].change.emit();\n'
+
+    slider_callback = CustomJS(
+        args=dict(figure=f, sources=out[0], frames=out),
+        code=js_string,
+    )
+
+    slider = Slider(start=0, end=len(frames), value=0, step=1, title="frame")
+    slider.js_on_change("value", slider_callback)
+
+    f.legend.location = "top_left"
+    f.legend.click_policy = "hide"
+
+    layout = column(f, slider)
+    if save_path is not None:
+        save(layout, filename=save_path, title=f"{scene_index}", resources=Resources(resources))
+    if do_show:
+        show(layout)
```

Comment:
I would remove this from here for the same reason as the `output_notebook()`, it should be the responsibility of the caller to decide what to do. We can just return the `layout` here and also remove the `resources` parameter from the function (that is also only being used here).